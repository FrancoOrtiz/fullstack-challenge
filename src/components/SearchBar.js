import React from 'react'

export const SearchBar = () => {
  return (
    <div>
        <input></input>
        <button></button>
    </div>
  )
}
