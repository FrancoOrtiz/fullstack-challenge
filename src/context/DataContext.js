import React, {createContext, useState, useEffect} from 'react'

export const DataContext = createContext();

export const DataProvider = ({children}) => {

    const [page, setPage] = useState(0);

    const [pokemons, setPokemons] = useState([]);

    const fetchAllPokemons = async () => {
        const url = `http://localhost:3001/api/pokemons/${page}`
        const peticion = await fetch(url)
        const pokemons = await peticion.json();
        setPokemons(pokemons);
    }

    useEffect(() => {
        fetchAllPokemons();
    }, [page])

    return (
        <DataContext.Provider value={{
            pokemons,
            setPokemons,
            fetchAllPokemons,
            setPage,
        }}>
            {children}
        </DataContext.Provider>
    )
}
