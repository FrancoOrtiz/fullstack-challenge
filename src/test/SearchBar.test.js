/* eslint-disable testing-library/prefer-screen-queries */
import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { SearchBar } from "../components/SearchBar.js";
import renderer from "react-test-renderer";
import { DataContext } from "../context/DataContext.js";

describe("Tests on SearchBar.js", () => {
  let wrapper;

  const mockByName = jest.fn();

  test("should be render", () => {
    wrapper = renderer
      .create(
        <DataContext.Provider value={{ pokemonByName: mockByName }}>
          <SearchBar />
        </DataContext.Provider>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  test("should be call pokemonByName function", () => {
    wrapper = render(
      <DataContext.Provider value={{ pokemonByName: mockByName }}>
        <SearchBar />
      </DataContext.Provider>
    );

    const btn = wrapper.getByText("Buscar");
    fireEvent.click(btn);

    expect(mockByName).toHaveBeenCalled();
  });

  test("should be called with the argument of input value", () => {
    wrapper = render(
      <DataContext.Provider value={{ pokemonByName: mockByName }}>
        <SearchBar></SearchBar>
      </DataContext.Provider>
    );

    const input = wrapper.getByPlaceholderText("Ingrese el nombre a buscar");
    fireEvent.change(input, { target: { value: "pikachu" } });

    const btn = wrapper.getByText("Buscar");
    fireEvent.click(btn);

    expect(mockByName).toHaveBeenCalledWith("pikachu");
  });
});
