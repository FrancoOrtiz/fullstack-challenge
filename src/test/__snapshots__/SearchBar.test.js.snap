// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on SearchBar.js should be render 1`] = `
<form
  className="w-full mt-6 flex items-center justify-between"
  onSubmit={[Function]}
>
  <input
    className="w-3/5 border-black border-2 pl-2 h-10 placeholder:text-black"
    name="name"
    onChange={[Function]}
    placeholder="Ingrese el nombre a buscar"
    type="text"
    value=""
  />
  <button
    className="w-1/5 py-3 shadow-button rounded-xl bg-gray-300 font-bold border-black border-2 text-xl"
    type="submit"
  >
    Buscar
  </button>
</form>
`;
